pkg/utils/provider_meta.go

```diff
@@ -59,6 +59,8 @@ func GetDBClientFromMeta(meta interface{}, d *schema.ResourceData) (*sqlx.DB, cl
 	var region clients.Region
 	if d != nil && d.Get("region") != "" {
 		region = clients.Region(d.Get("region").(string))
+	} else if d != nil && ExtractRegion(d.Id()) != "" {
+		region = clients.Region(ExtractRegion(d.Id()))
 	} else {
 		region = providerMeta.DefaultRegion
 	}
```

Comment:
There was a bug here which only manifested when importing resources with a non-default region.

During the import stage, we essentially only have access to the ID of the specific resource, eg:

```
terraform import materialize_role.example_role <region>:<role_id>
```

However we never checked the old logic was only checking for the `region` attribute which is not available during the import stage, and else it was using the default region, which might have been different from the region of the imported resource.